include/picongpu/simulationControl/AdjustDomainSize.hpp

```diff
@@ -0,0 +1,363 @@
+            if( validLocalSize != m_localDomainSize[ dim ] )
+            {
+                std::cout << "Dimension " << dim <<
+                    ": local grid size is not contains at least 3 supercells. Adjust from " <<
+                    m_localDomainSize[ dim ] << " to " <<  validLocalSize << std::endl;
+                sizeChanged = true;
+            }
+
+            m_localDomainSize[ dim ] = validLocalSize;
+        }
+
+        /** ensure that the absorber fits into the local domain
+         *
+         * The methods checks the local domain size only if the absorber for the
+         * given dimension is enabled.
+         *
+         * @param dim dimension to update
+         */
+        void greaterEqualThanAbsorber( size_t const dim )
+        {
+            int validLocalSize = m_localDomainSize[ dim ];
+            if(
+                !m_isPeriodic[ dim ] &&
+                // device has absorber
+                ( m_mpiPosition[ dim ] == 0 || m_mpiPosition[ dim ] == m_numDevices[ dim ] - 1 )
+            )
+            {
+                int const maxAbsorberCells = static_cast< int >(
+                    std::max(
+                        ABSORBER_CELLS[ dim ][ 0 ],
+                        ABSORBER_CELLS[ dim ][ 1 ]
+                    )
+                );
+
+                if( m_localDomainSize[ dim ] < maxAbsorberCells )
+                {
+                    int const sCellSize = SuperCellSize::toRT()[ dim ];
+                    // round up to full supercells
+                    validLocalSize =
+                        ( ( maxAbsorberCells + sCellSize - 1 ) / sCellSize ) *
+                        sCellSize;
+                }
+
+                if( validLocalSize != m_localDomainSize[ dim ] )
+                {
+                    std::cout << "Dimension " << dim <<
+                        ": local grid size must be greater or equal than the largest absorber. Adjust from " <<
+                        m_localDomainSize[ dim ] << " to " <<  validLocalSize << std::endl;
+                    sizeChanged = true;
+                }
+
+                m_localDomainSize[ dim ] = validLocalSize;
+            }
+        }
+
+        /** derive the local domain size
+         *
+         * Calculate the local domain size.
+         * This function take into account that the local domain size must be
+         * equal for all domains if moving window is activated.
+         *
+         * @param dim dimension to update
+         */
+        void deriveLocalDomainSize( size_t const dim )
+        {
+            if( m_movingWindowEnabled && dim == 1 )
+            {
+
+                pmacc::mpi::MPIReduce mpiReduce;
+
+                int globalMax;
+                mpiReduce(
+                    pmacc::nvidia::functors::Max(),
+                    &globalMax,
+                    &m_localDomainSize[ dim ],
+                    1,
+                    pmacc::mpi::reduceMethods::AllReduce()
```

Comment:
Out of curiosity, why not use the version of this method without the last parameter?

Reply:
I'm not sure either, hence asked.